apps/errors/error_design.md

```diff
@@ -0,0 +1,47 @@
+# MSAL Error Design
```

Comment:
it should be easy for the consumer of the library to know what they need to do based on the error returned (exception or not). The client exceptions have one purpose, service exceptions has another purpose. also we need to support classifications to ensure we can follow standard MSAL patterns.

Reply:
@henrik-me 

This should do all of that.  You can diff on if the error is a service error with the built in errors.Is().  All errors contain error text on what the issue is.  Service errors come with not only the error but a Verbose(err) to yank out the http request and response object.

If we need to add specific sub-errors that can be switched on, we can do that with the same error typing we do with service errors and they can be detected with errors.Is() as well.  

I would say that the difficulty for the consumer will be in interpreting what the server side is saying, as those errors are not always clear.